docs/content/en/docs/triggers/message-queue-trigger/message-queue-trigger-kind-keda.md

```diff
@@ -0,0 +1,46 @@
+---
+title: "MQ Trigger of kind Keda"
+date: 2020-07-20T14:30:01+05:30
+weight: 1
+---
+
+# Brief Introduction
+
+Message queue trigger integration with KEDA has enabled autoscaling of trigger handler. Now, there are two kinds of message queue triggers:
+
+1. fission
+2. keda
+
+Message queue trigger kind can be specified using "mqtkind" flag. By default, "mqtkind" is set to "fission" which creates regular [message queue trigger](https://docs.fission.io/docs/triggers/message-queue-trigger/). To create message queue trigger of kind keda one must specify "mqtkind=keda".
+
+# Architecture
+
+{{< img "../assets/mqt-kind-keda.png" "" "40em" "1" >}}
+
+1. The user creates a trigger - for Keda based integration you have to specify the “mqtkind=keda” and add all relevant parameters. These parameters are different for each message queue and hence are encapsulated in a metadata field and follow a key-value format. As soon as you create the MQ Trigger, Fission creates a ScaledObject and a consumer deployment object which is referenced by ScaledObject. The ScaledObject is a Keda’s way of encapsulating the consumer deployment and all relevant information for connecting to an event source! Keda goes ahead and creates a HPA for the deployment and scales down the deployment to zero.
+2. As the message arrives in the event source - the Keda will scale the HPA and deployment from 0 - to 1 for consuming messages. As more messages arrive the deployment is scaled beyond 1 automatically too.
+3. The deployment is like an connector which consumes messages from the source and then calls a function.
+4. The function consumes the message and returns the response to deployment pods, which puts the response in response topic and errors in error topic as may be applicable.
+
+# Usage
+
```

Comment:
1. A small note about Keda need to be installed
2. How to enable Keda integration in Fission by changing values.yaml?

Reply:
Done